### ISSUE TITLE
AVA Support

### ISSUE THREAD

**Issue by JbIPS** [mxsdev/nvim-dap-vscode-js#24]
Hi, I'm trying to run AVA test with DAP.

I tried the [example configuration](https://github.com/avajs/ava/blob/main/docs/recipes/debugging-with-vscode.md) in the documentation. It works well with VSCode but in NVim it starts the program but do not stop on breakpoints.

I also tried a *generic* config with `npm test`:
```lua
{
    type = "pwa-node",
    request = "launch",
    name = "Run NPM test",
    cwd = "${workspaceFolder}",
    runtimeExecutable = "npm",
    runtimeArgs = {
      "test"
    },
    trace = true,
    outputCapture = "std",
    console = "integratedTerminal",
    outFiles = {
      "${workspaceFolder}/dist/*.js",
      "!**/node_modules/**"
    }
  }
```
but I got the same results.

I tried to add a `--inspect` in the runtimeArgs but I get an error because the inspector is already started. Again, this config works well with VSCode.

Can you point me to a difference between VSCode and DAP that can leads to this?

Thank you

**Comment by anasrar** [comment]
Work break point and jump break point.
```lua
    {
      type = "pwa-node",
      request = "launch",
      name = "Launch Test Current File (pwa-node with ava)",
      cwd = vim.fn.getcwd(),
      program = "${workspaceFolder}/node_modules/ava/entrypoints/cli.mjs",
      args = { "--inspect-brk", "${file}" },
      autoAttachChildProcesses = true,
      smartStep = true,
      console = "integratedTerminal",
      skipFiles = { "<node_internals>/**", "node_modules/**" },
    },
```

![image](https://user-images.githubusercontent.com/38805204/211494329-fdbe9f70-505d-4434-af06-d8e13fb14c44.png)

**Comment by JbIPS** [comment]
I've got the same result with this configuration.

I'm sorry I just realized I missed a major point: I'm using TypeScript. The tests are precompiled and a remapping is done with [AVA TypeScript](https://github.com/avajs/typescript).

Maybe there's something with the source maps?

**Comment by mxsdev** [comment]
Hi!

I'm not super familiar with AVA unfortunately. To confirm, is `npx ava debug test.ts` a total solution?

Regardless, if it works in VSCode but not in this plugin, I'm hoping that #15 will resolve any inconsistencies, since technically this plugin uses a different (probably lesser-maintained) endpoint to the debugger than VSCode. That issue is currently blocked on Microsoft/nvim-dap, I believe.

**Comment by JbIPS** [comment]
I hope this will be resolved with #15 :crossed_fingers: 

To be clear, the debugger can attach itself to a `npx ava debug test.ts` process with the default `Attach` configuration. I'm not skilled enough to understand what goes under the hood in a `Attach` vs `Launch`, so I'll wait for `startDebuging` and continue with my workaround until then.

Thank you for your help and this plugin 🙂

**Comment by JbIPS** [comment]
Great news today, after a lot of updates and tweaks I finally made it work!

Here's my config for future reference:
```lua
  {
    type = "pwa-node",
    request = "launch",
    name = "Debug AVA Test file",
    program = "node_modules/ava/entrypoints/cli.mjs",
    args = {
      '--serial',
      '${file}'
    },
    cwd = "${workspaceFolder}",
    outFiles = {"${workspaceFolder}/dist/**/*.js"},
    console = "integratedTerminal"
  },
```

Ava config: 
```js
export default {
  typescript: {
    rewritePaths: {
      'src/': 'dist/'
    },
    compile: false
  }
}
```
